notebooks/Tutorial_Top-K_Multidimensional_Motif_and_Matches_Discovery.ipynb

```diff
@@ -0,0 +1,827 @@
+{
```

Comment:
<p>This is actually pretty cool! However, shouldn't we be able to capture the spikes in the "Washing Machine" usage? It's hard to tell without zooming in but it feels like we're only capturing the noise?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/557/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Reply:
<p> </p><p>Yes, I was also thinking about this. But I'm not sure what to do about the noise problem. Do you think that increasing the amount of noise would change anything? Do you have any ideas?</p>

Reply:
> By any chance, have you ever tried:
> 
>     Not adding noise
>     Replace zero regions with np.nan
>     Use a variety of smaller window sizes

Yes, I was trying to do so, but it seems like as if it was eliminating too much :(

> Additionally, I think you had mentioned this before but it does sound like there may be an annotation vector play here. What's challenging is that, from the multi-dimensional matrix profile (aka "multi-MP"), we don't immediately know which dimensions contributed to each distance value (in the multi-MP). So, we would need to do something like:
> 
>     Start with the multi-MP
>     For each value, determine the precise subspace that generated the distance value (we know what the precise subsequence pairs are)
>     For a given matrix profile distance and subspace, determine if the (multi-dimensional) subsequences within that subspace contain zero regions
>     if so, set the multi-MP distance value for that dimension and subsequence index to np.inf
> 
> Essentially, in steps 1-3, we're generating a multi-dimensional annotation vector. I think this is a bit of work to do but it may actually be a process flow that I think users will inevitably encounter with their real-world data. Would you care to play this?

Yeah I think it may be worth investigating some time into it and see if it is working as expected. However, I am currently quite stressed with the writing of the bachelor thesis, because my program does not yet want the way that I would like it to be and I also have to evaluate quite a lot. It may therefore take some time until I can look at the problem. But I'll definitely check it out as soon as I have some more free time again!

Reply:
> It may therefore take some time until I can look at the problem. But I'll definitely check it out as soon as I have some more free time again!

Please take your time! I'll try to offer some code suggestions related to the three points above.

Reply:
All right, sounds good :)

Reply:
Ahh, cool, thank you very much! Unfortunately I'm very stressed with my thesis and final presentation at the moment. There is only one month left and afterwards I have got it! Then I'll have much time to think about this and try different things out. If you want to, I could then also help you with the one-dimensional motif and matches tutorial that you wanted to create or help with something different :)

Reply:
<p>That would be great! Best of luck</p>

Reply:
Hey :)

I finally finished my bachelor thesis. I'll try to integrate your suggested Annotation Vector approach the days!

Reply:
<p>Welcome back and congratulations on wrapping up your bachelors thesis! </p>